### DISCUSSION TITLE
[Question] password hashing

### DISCUSSION THREAD

**Discussion by s-celles** [flet-dev/flet#1895] [Q&A]
### Question

Hello,

I noticed this https://flet.dev/docs/guides/python/encrypting-sensitive-data/ article which explains how to cipher password using flet encrypt / decrypt function

I think this article should explain how to deal with a .env file to store secret.

Moreover I don't think it's very safe to store passwords in application / db even a ciphered version of password.

Hashing a salted password may be prefered.
https://www.google.com/search?q=hash+salt+password

Any opinion about that?

Kind regards

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by s-celles** [comment]
Here is how I processed

```
import os
import bcrypt


def hash_password(login: str, password: str):
    # salt = bcrypt.gensalt()
    salt = os.getenv("MYAPP_SECRET_KEY")
    return bcrypt.hashpw((password + login).encode(), salt.encode())

def check_password(login: str, password: str, hashed_password: bytes):
    new_hashed_password = hash_password(login, password)
    return new_hashed_password == hashed_password
```

**Reply by FeodorFitsner** [reply]
Would you like to do a PR to that doc article?

**Reply by plugeit** [reply]
The best thing would be store the key on a server like Firebase and retrieve it from their added security

**Reply by s-celles** [reply]
I will tackle that if someone can help me on https://github.com/flet-dev/flet/discussions/1891

**Reply by plugeit** [reply]
my app is going to have only otp login
